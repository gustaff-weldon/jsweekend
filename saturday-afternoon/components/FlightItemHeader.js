// @flow

import * as React from "react";
import { Row, Col } from "antd";

const FlightItemHeader = () => (
  <Row type="flex" justify="space-between">
    <Col>
      <img
        src="https://images.kiwi.com/airlines/64/S7.png"
        width={32}
        height={32}
      />
    </Col>
    <Col>Azul, Norwegian</Col>
    <Col>
      <Row>
        <time datetime="2018-03-18T10:25:00+00:00">10:25</time>
        <span> - </span>
        <time datetime="2018-03-18T15:10:00+00:00">15:10</time>
      </Row>
      <Row>
        <time datetime="2018-03-18T10:25:00+00:00">Sun 18 Mar</time>
      </Row>
    </Col>
    <Col>Prague (PRG) - Bangkok (BKK)</Col>
    <Col>
      <div>120 EUR</div>
    </Col>
  </Row>
);

export default FlightItemHeader;
