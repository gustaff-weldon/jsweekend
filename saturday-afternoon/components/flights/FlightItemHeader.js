// @flow

import * as React from "react";
import { Row, Col } from "antd";
import moment from "moment";

type Props = {
  flight: Flight
};

type Flight = {
  departure: Object,
  arrival: Object,
  duration: Object,
  legs: Object,
  price: Object
};

const FlightItemHeader = ({
  flight: { departure, arrival, duration, legs, price }
}: Props) => {
  const durationFormatted = moment.duration(duration, "minutes");

  return (
    <div>
      <Row>
        <Col span={6}>
          <Row>
            {legs.map(leg => {
              return (
                <Col key={leg.id} style={{ marginBottom: 10 }}>
                  <img
                    src={leg.airline.logoUrl}
                    alt={leg.airline.name}
                    width={32}
                    height={32}
                    style={{ marginRight: 10 }}
                  />
                  <span>{leg.airline.name}</span>
                </Col>
              );
            })}
          </Row>
        </Col>
        <Col span={4}>
          <Row>
            <time dateTime={departure.time}>
              {moment(departure.time).format("HH:MM")}
            </time>
            <span> - </span>
            <time dateTime={arrival.time}>
              {moment(arrival.time).format("HH:MM")}
            </time>
          </Row>
          <Row>
            <time dateTime={departure.time}>
              {moment(departure.time).format("dd MMM D")}
            </time>
          </Row>
        </Col>
        <Col span={4}>
          {durationFormatted.hours()}h {durationFormatted.minutes()}m
        </Col>
        <Col span={6}>
          {departure.airport.city.name} ({departure.airport.locationId}) -{" "}
          {arrival.airport.city.name} ({arrival.airport.locationId})
        </Col>
        <Col span={4}>
          <div className="price">
            {price.amount} {price.currency}
          </div>
        </Col>
      </Row>
      <style jsx>{`
        .price {
          text-align: right;
        }
      `}</style>
    </div>
  );
};

export default FlightItemHeader;
