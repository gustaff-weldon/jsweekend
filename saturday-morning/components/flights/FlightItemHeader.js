// @flow

import * as React from "react";
import { Row, Col } from "antd";
import moment from "moment";

type Props = {
  flight: Flight
};

type Flight = {
  departure: Object,
  arrival: Object,
  duration: number,
  legs: Leg[],
  price: Object
};

type Leg = {
  id: string,
  airline: {
    name: string,
    logoUrl: string
  },
  arrival: {
    time: string,
    airport: {
      name: string,
      city: {
        name: string
      }
    }
  },
  departure: {
    time: string,
    airport: {
      name: string,
      city: {
        name: string
      }
    }
  }
};

const FlightItemHeader = ({
  flight: { departure, arrival, duration, legs, price }
}: Props) => {
  const durationFormatted = moment.duration(duration, "minutes");

  return (
    <Row type="flex" justify="space-between">
      <Row>
        {legs.map(leg => {
          return (
            <Col key={leg.id} style={{ marginBottom: 10 }}>
              <img
                src={leg.airline.logoUrl}
                alt={leg.airline.name}
                width={32}
                height={32}
                style={{ marginRight: 10 }}
              />
              <span>{leg.airline.name}</span>
            </Col>
          );
        })}
      </Row>
      <Col>
        <Row>
          <time dateTime={departure.time}>
            {moment.utc(departure.time).format("HH:mm")}
          </time>
          <span> - </span>
          <time dateTime={arrival.time}>
            {moment.utc(arrival.time).format("HH:mm")}
          </time>
        </Row>
        <Row>
          <time dateTime={departure.time}>
            {moment.utc(departure.time).format("dd MMM D")}
          </time>
        </Row>
      </Col>
      <Col>
        {durationFormatted.hours()}h {durationFormatted.minutes()}m
      </Col>
      <Col>
        {departure.airport.city.name} ({departure.airport.locationId}) -{" "}
        {arrival.airport.city.name} ({arrival.airport.locationId})
      </Col>
      <Col>
        <div>
          {price.amount} {price.currency}
        </div>
      </Col>
    </Row>
  );
};

export default FlightItemHeader;
