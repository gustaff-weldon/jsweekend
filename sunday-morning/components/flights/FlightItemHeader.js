// @flow

import * as React from "react";
import { createFragmentContainer, graphql } from "react-relay";
import { Row, Col } from "antd";
import moment from "moment";

type Props = {
  flight: Flight
};

type Flight = {
  departure: Object,
  arrival: Object,
  duration: Object,
  legs: Object,
  price: Object
};

const FlightItemHeader = ({
  flight: { departure, arrival, duration, legs, price }
}: Props) => {
  const durationFormatted = moment.duration(duration, "minutes");

  return (
    <div>
      <Row>
        <Col span={6}>
          <Row>
            {legs.map(leg => {
              return (
                <Col key={leg.id} style={{ marginBottom: 10 }}>
                  <img
                    src={leg.airline.logoUrl}
                    alt={leg.airline.name}
                    width={32}
                    height={32}
                    style={{ marginRight: 10 }}
                  />
                  <span>{leg.airline.name}</span>
                </Col>
              );
            })}
          </Row>
        </Col>
        <Col span={4}>
          <Row>
            <time dateTime={departure.localTime}>
              {moment.utc(departure.localTime).format("HH:mm")}
            </time>
            <span> - </span>
            <time dateTime={arrival.localTime}>
              {moment.utc(arrival.localTime).format("HH:mm")}
            </time>
          </Row>
          <Row>
            <time dateTime={departure.localTime}>
              {moment.utc(departure.localTime).format("dd MMM D")}
            </time>
          </Row>
        </Col>
        <Col span={4}>
          {durationFormatted.hours()}h {durationFormatted.minutes()}m
        </Col>
        <Col span={6}>
          {departure.airport.city.name} ({departure.airport.locationId}) -{" "}
          {arrival.airport.city.name} ({arrival.airport.locationId})
        </Col>
        <Col span={4}>
          <div className="price">
            {price.amount} {price.currency}
          </div>
        </Col>
      </Row>
      <style jsx>{`
        .price {
          text-align: right;
        }
      `}</style>
    </div>
  );
};

export default createFragmentContainer(
  FlightItemHeader,
  graphql`
    fragment FlightItemHeader_flight on Flight {
      legs {
        id
        airline {
          name
          logoUrl
        }
      }
      departure {
        localTime
        airport {
          locationId
          city {
            name
          }
        }
      }
      arrival {
        localTime
        airport {
          locationId
          city {
            name
          }
        }
      }
      duration
      price {
        amount
        currency
      }
    }
  `
);
