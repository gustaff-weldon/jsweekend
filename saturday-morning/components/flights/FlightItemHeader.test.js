// @flow

import React from "react";
import { shallow } from "enzyme";

import FlightItemHeader from "./FlightItemHeader";

const data = {
  id: "RmxpZ2h0OjM2NTI1Nzc5NzgyMDI2NjlfMHwzNjExODk2NDM2MTYxMzI5XzA=",
  departure: {
    localTime: "2018-03-16T07:40:00.000Z",
    airport: { locationId: "PRG", city: { name: "Prague" } }
  },
  arrival: {
    localTime: "2018-03-16T15:45:00.000Z",
    airport: { locationId: "BCN", city: { name: "Barcelona" } }
  },
  duration: 485,
  legs: [
    {
      id: "TGVnOjM2NTI1Nzc5NzgyMDI2NjlfMA==",
      airline: {
        name: "Ryanair",
        logoUrl: "https://images.kiwi.com/airlines/64/FR.png"
      }
    },
    {
      id: "TGVnOjM2MTE4OTY0MzYxNjEzMjlfMA==",
      airline: {
        name: "Ryanair",
        logoUrl: "https://images.kiwi.com/airlines/64/FR.png"
      }
    }
  ],
  price: { amount: 130, currency: "EUR" }
};

describe("FlightItemHeader", () => {
  it("should match snapshot", () => {
    expect(shallow(<FlightItemHeader flight={data} />)).toMatchSnapshot();
  });
});
