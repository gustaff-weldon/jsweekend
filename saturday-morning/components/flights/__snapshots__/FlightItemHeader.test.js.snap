// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightItemHeader should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FlightItemHeader
    flight={
      Object {
        "arrival": Object {
          "airport": Object {
            "city": Object {
              "name": "Barcelona",
            },
            "locationId": "BCN",
          },
          "localTime": "2018-03-16T15:45:00.000Z",
        },
        "departure": Object {
          "airport": Object {
            "city": Object {
              "name": "Prague",
            },
            "locationId": "PRG",
          },
          "localTime": "2018-03-16T07:40:00.000Z",
        },
        "duration": 485,
        "id": "RmxpZ2h0OjM2NTI1Nzc5NzgyMDI2NjlfMHwzNjExODk2NDM2MTYxMzI5XzA=",
        "legs": Array [
          Object {
            "airline": Object {
              "logoUrl": "https://images.kiwi.com/airlines/64/FR.png",
              "name": "Ryanair",
            },
            "id": "TGVnOjM2NTI1Nzc5NzgyMDI2NjlfMA==",
          },
          Object {
            "airline": Object {
              "logoUrl": "https://images.kiwi.com/airlines/64/FR.png",
              "name": "Ryanair",
            },
            "id": "TGVnOjM2MTE4OTY0MzYxNjEzMjlfMA==",
          },
        ],
        "price": Object {
          "amount": 130,
          "currency": "EUR",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Col
          span={6}
        >
          <Airlines
            legs={
              Array [
                Object {
                  "airline": Object {
                    "logoUrl": "https://images.kiwi.com/airlines/64/FR.png",
                    "name": "Ryanair",
                  },
                  "id": "TGVnOjM2NTI1Nzc5NzgyMDI2NjlfMA==",
                },
                Object {
                  "airline": Object {
                    "logoUrl": "https://images.kiwi.com/airlines/64/FR.png",
                    "name": "Ryanair",
                  },
                  "id": "TGVnOjM2MTE4OTY0MzYxNjEzMjlfMA==",
                },
              ]
            }
          />
        </Col>,
        <Col
          span={4}
        >
          <Row
            gutter={0}
          >
            <DateTimeRange
              arrivalTime="2018-03-16T15:45:00.000Z"
              departureTime="2018-03-16T07:40:00.000Z"
            />
          </Row>
          <Row
            gutter={0}
          >
            <DateTime
              isDate={true}
              value="2018-03-16T07:40:00.000Z"
            />
          </Row>
        </Col>,
        <Col
          span={4}
        >
          <Duration
            value={485}
          />
        </Col>,
        <Col
          span={6}
        >
          Prague
           (
          PRG
          ) -
           
          Barcelona
           (
          BCN
          )
        </Col>,
        <Col
          span={4}
        >
          130
           
          EUR
        </Col>,
      ],
      "gutter": 0,
      "justify": "space-between",
      "type": "flex",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Airlines
            legs={
              Array [
                Object {
                  "airline": Object {
                    "logoUrl": "https://images.kiwi.com/airlines/64/FR.png",
                    "name": "Ryanair",
                  },
                  "id": "TGVnOjM2NTI1Nzc5NzgyMDI2NjlfMA==",
                },
                Object {
                  "airline": Object {
                    "logoUrl": "https://images.kiwi.com/airlines/64/FR.png",
                    "name": "Ryanair",
                  },
                  "id": "TGVnOjM2MTE4OTY0MzYxNjEzMjlfMA==",
                },
              ]
            }
          />,
          "span": 6,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "legs": Array [
              Object {
                "airline": Object {
                  "logoUrl": "https://images.kiwi.com/airlines/64/FR.png",
                  "name": "Ryanair",
                },
                "id": "TGVnOjM2NTI1Nzc5NzgyMDI2NjlfMA==",
              },
              Object {
                "airline": Object {
                  "logoUrl": "https://images.kiwi.com/airlines/64/FR.png",
                  "name": "Ryanair",
                },
                "id": "TGVnOjM2MTE4OTY0MzYxNjEzMjlfMA==",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Row
              gutter={0}
            >
              <DateTimeRange
                arrivalTime="2018-03-16T15:45:00.000Z"
                departureTime="2018-03-16T07:40:00.000Z"
              />
            </Row>,
            <Row
              gutter={0}
            >
              <DateTime
                isDate={true}
                value="2018-03-16T07:40:00.000Z"
              />
            </Row>,
          ],
          "span": 4,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <DateTimeRange
                arrivalTime="2018-03-16T15:45:00.000Z"
                departureTime="2018-03-16T07:40:00.000Z"
              />,
              "gutter": 0,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "arrivalTime": "2018-03-16T15:45:00.000Z",
                "departureTime": "2018-03-16T07:40:00.000Z",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <DateTime
                isDate={true}
                value="2018-03-16T07:40:00.000Z"
              />,
              "gutter": 0,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "isDate": true,
                "value": "2018-03-16T07:40:00.000Z",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Duration
            value={485}
          />,
          "span": 4,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "value": 485,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Prague",
            " (",
            "PRG",
            ") -",
            " ",
            "Barcelona",
            " (",
            "BCN",
            ")",
          ],
          "span": 6,
        },
        "ref": null,
        "rendered": Array [
          "Prague",
          " (",
          "PRG",
          ") -",
          " ",
          "Barcelona",
          " (",
          "BCN",
          ")",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            130,
            " ",
            "EUR",
          ],
          "span": 4,
        },
        "ref": null,
        "rendered": Array [
          130,
          " ",
          "EUR",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Col
            span={6}
          >
            <Airlines
              legs={
                Array [
                  Object {
                    "airline": Object {
                      "logoUrl": "https://images.kiwi.com/airlines/64/FR.png",
                      "name": "Ryanair",
                    },
                    "id": "TGVnOjM2NTI1Nzc5NzgyMDI2NjlfMA==",
                  },
                  Object {
                    "airline": Object {
                      "logoUrl": "https://images.kiwi.com/airlines/64/FR.png",
                      "name": "Ryanair",
                    },
                    "id": "TGVnOjM2MTE4OTY0MzYxNjEzMjlfMA==",
                  },
                ]
              }
            />
          </Col>,
          <Col
            span={4}
          >
            <Row
              gutter={0}
            >
              <DateTimeRange
                arrivalTime="2018-03-16T15:45:00.000Z"
                departureTime="2018-03-16T07:40:00.000Z"
              />
            </Row>
            <Row
              gutter={0}
            >
              <DateTime
                isDate={true}
                value="2018-03-16T07:40:00.000Z"
              />
            </Row>
          </Col>,
          <Col
            span={4}
          >
            <Duration
              value={485}
            />
          </Col>,
          <Col
            span={6}
          >
            Prague
             (
            PRG
            ) -
             
            Barcelona
             (
            BCN
            )
          </Col>,
          <Col
            span={4}
          >
            130
             
            EUR
          </Col>,
        ],
        "gutter": 0,
        "justify": "space-between",
        "type": "flex",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Airlines
              legs={
                Array [
                  Object {
                    "airline": Object {
                      "logoUrl": "https://images.kiwi.com/airlines/64/FR.png",
                      "name": "Ryanair",
                    },
                    "id": "TGVnOjM2NTI1Nzc5NzgyMDI2NjlfMA==",
                  },
                  Object {
                    "airline": Object {
                      "logoUrl": "https://images.kiwi.com/airlines/64/FR.png",
                      "name": "Ryanair",
                    },
                    "id": "TGVnOjM2MTE4OTY0MzYxNjEzMjlfMA==",
                  },
                ]
              }
            />,
            "span": 6,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "legs": Array [
                Object {
                  "airline": Object {
                    "logoUrl": "https://images.kiwi.com/airlines/64/FR.png",
                    "name": "Ryanair",
                  },
                  "id": "TGVnOjM2NTI1Nzc5NzgyMDI2NjlfMA==",
                },
                Object {
                  "airline": Object {
                    "logoUrl": "https://images.kiwi.com/airlines/64/FR.png",
                    "name": "Ryanair",
                  },
                  "id": "TGVnOjM2MTE4OTY0MzYxNjEzMjlfMA==",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Row
                gutter={0}
              >
                <DateTimeRange
                  arrivalTime="2018-03-16T15:45:00.000Z"
                  departureTime="2018-03-16T07:40:00.000Z"
                />
              </Row>,
              <Row
                gutter={0}
              >
                <DateTime
                  isDate={true}
                  value="2018-03-16T07:40:00.000Z"
                />
              </Row>,
            ],
            "span": 4,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <DateTimeRange
                  arrivalTime="2018-03-16T15:45:00.000Z"
                  departureTime="2018-03-16T07:40:00.000Z"
                />,
                "gutter": 0,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "arrivalTime": "2018-03-16T15:45:00.000Z",
                  "departureTime": "2018-03-16T07:40:00.000Z",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <DateTime
                  isDate={true}
                  value="2018-03-16T07:40:00.000Z"
                />,
                "gutter": 0,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "isDate": true,
                  "value": "2018-03-16T07:40:00.000Z",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Duration
              value={485}
            />,
            "span": 4,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "value": 485,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Prague",
              " (",
              "PRG",
              ") -",
              " ",
              "Barcelona",
              " (",
              "BCN",
              ")",
            ],
            "span": 6,
          },
          "ref": null,
          "rendered": Array [
            "Prague",
            " (",
            "PRG",
            ") -",
            " ",
            "Barcelona",
            " (",
            "BCN",
            ")",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              130,
              " ",
              "EUR",
            ],
            "span": 4,
          },
          "ref": null,
          "rendered": Array [
            130,
            " ",
            "EUR",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
